<launch>
    <arg name="debug"               default="false"/>
    <arg name="bag_filename"        default="$(env HOME)/bags/campusrover_bag/20200529_elevator_1.bag"/>
    <arg name="elevator_path"       default="/elevator_path"/>
    <arg name="global_path"         default="/global_path"/>
    <arg name="local_costmap"       default="/campusrover_local_costmap"/>
    <arg name="cmd_vel"             default="/yocs_cmd_vel_mux/input/navigation"/>
    <!-- /////////////////////////////////////////////////////////////////////////////////// -->
    <group if="$(arg debug)">

        <param name="use_sim_time" value="true" />
        <node name="playbag" pkg="rosbag" type="play"
                args="--clock $(arg bag_filename)" />

        <param name="robot_description"
            textfile="$(find campusrover_description)/urdf/campusrover.urdf" />

        <node name="robot_state_publisher" pkg="robot_state_publisher"
            type="robot_state_publisher" />


        <!-- rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find campusrover_move)/rviz/dp_planner.rviz" required="true"/>

    </group>

    <node pkg="campusrover_move" type="path_following" name="path_following" output="screen">
        <remap from="elevator_path"    to="$(arg elevator_path)" />
        <remap from="global_path"    to="$(arg global_path)" />
        <remap from="costmap"    to="$(arg local_costmap)" />
        <remap from="cmd_vel"    to="$(arg cmd_vel)" />
        <param name="robot_frame" value="base_link" />

        <param name="arriving_range_dis" value="0.2" />
        <param name="arriving_range_angle" value="0.1" />
        
        <param name="max_linear_velocity" value="0.5" />
        <param name="max_angular_velocity" value="0.4" />
        <param name="target_point_dis" value="0.6" />

        <param name="threshold_occupied" value="10" />
        <param name="footprint_max_x" value="0.4" />
        <param name="footprint_min_x" value="-0.1" />
        <param name="footprint_max_y" value="0.01" />
        <param name="footprint_max_y" value="-0.03" />

        <param name="speed_pid_k" value="0.5" />

        <param name="enble_costmap_obstacle" value="false" />
        <param name="direction_inverse" value="false" />
    </node>
</launch>